src/typed.jl

```diff
@@ -68,8 +69,12 @@ function dispatch_msg(x::JSONRPCEndpoint, dispatcher::MsgDispatcher, msg)
             if handler.message_type isa RequestType
                 if res isa JSONRPCError
                     send_error_response(x, msg, res.code, res.msg, res.data)
-                else
+                elseif res isa get_return_type(handler.message_type)
```

Comment:
Defining `return_type` field in `Handler` struct and using it instead of dispatching every time would be better ?
Inner constructor would be useful for this case, I guess.

Reply:
Yes, we could do this in a future iteration, but I think for now I just want to get this merged, and this works :)